.travis.yml

```diff
@@ -1,30 +1,115 @@
+# Based on the "trust" template v0.1.1
+# https://github.com/japaric/trust/tree/v0.1.1
+
+dist: trusty
 language: rust
-sudo: false
-before_script:
-  - pip install 'travis-cargo<0.2' --user && export PATH=$HOME/.local/bin:$PATH
+services: docker
+sudo: required
+
+# TODO Rust builds on stable by default, this can be
+# overridden on a case by case basis down below.
+
+env:
+  global:
+    # TODO Update this to match the name of your project.
+    - CRATE_NAME=trust
 
 matrix:
+  # TODO These are all the build jobs. Adjust as necessary. Comment out what you
+  # don't need
   include:
-    - rust: 1.18.0
-    - rust: stable
-    - rust: stable
+    # Android
+    # - env: TARGET=aarch64-linux-android DISABLE_TESTS=1
+    # - env: TARGET=arm-linux-androideabi DISABLE_TESTS=1
+    - env: TARGET=armv7-linux-androideabi DISABLE_TESTS=1
+    # - env: TARGET=i686-linux-android DISABLE_TESTS=1
+    # - env: TARGET=x86_64-linux-android DISABLE_TESTS=1
+
+    # iOS
+    # - env: TARGET=aarch64-apple-ios DISABLE_TESTS=1
+    #   os: osx
+    - env: TARGET=armv7-apple-ios DISABLE_TESTS=1
       os: osx
-    - rust: beta
-    - rust: nightly
-      script:
-        - cargo test --all
-        - cargo test --all --features nightly
-        - cargo test --all --benches
-        - cargo build --all --no-default-features
-        - cargo doc --no-deps --features nightly
```

Comment:
I guess we can do without the doc build. It would be nice to have the benchmark test (build test) because I've accidentally broken those a couple of times, but that can only be built on nightly. If you're not sure how just leave a comment somewhere.

Reply:
Yes, I have broken benches the too...

In `CI/script.sh` I have the tests:
```
cross build --target $TARGET
cross build --all --no-default-features --target $TARGET --release
cross test --all --target $TARGET
```

And for nightly only:
```
cross doc --no-deps --features nightly
cross test --all --features nightly --target $TARGET
cross test --all --benches --target $TARGET
```

So we shouldn't be missing something big.